odb-compiler/templates/config.hpp.in

```diff
@@ -15,6 +15,17 @@
 #cmakedefine ODBC_VERBOSE_BISON
 #cmakedefine ODBC_VERBOSE_FLEX
 
+#define ODBC_WIN32 0
+#define ODBC_MACOS 1
+#define ODBC_LINUX 2
+#if defined(__WIN32__) || defined(_WIN32)
+#define ODBC_PLATFORM ODBC_WIN32
+#elif defined(__APPLE_CC__)
+#define ODBC_PLATFORM ODBC_MACOS
+#elif defined(__linux__) && !defined(__ANDROID__)
+#define ODBC_PLATFORM ODBC_LINUX
+#endif
```

Comment:
I still feel like this is really ugly. If we're going to use defines I think we should at least do:
```cpp
#define ODBC_PLATFORM_${ODBC_PLATFORM}
```
in ```config.hpp.in``` and then use cmake to set ```ODBC_PLATFORM``` appropriately.

That way you can't make the mistake of doing
```cpp
#ifdef ODBC_PLATFORM
// whoops
#endif
```
and additionally, the value of ODBC_PLATFORM can't ever change because it is set at configure time instead of at compile time (in theory you could find a compiler that doesn't define _WIN32 or __WIN32__ and that would change how clients can use the library)

Reply:
That's fine with me. Was wondering whether to go with the configure approach or not.